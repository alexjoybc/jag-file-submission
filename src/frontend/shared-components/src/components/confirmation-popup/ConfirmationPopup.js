import React from "react";
import PropTypes from "prop-types";
import Modal from "react-bootstrap/Modal";
import { Button } from "../button/Button";
import { propTypes } from "../../types/propTypes";

import "./ConfirmationPopup.css";

export default function ConfirmationPopup({
  modal: { show, handleShow, handleClose, handleConfirm, title, body },
  mainButton,
  confirmButton,
  cancelButton
}) {
  return (
    <>
      <Button
        onClick={handleShow}
        label={mainButton.label}
        styling={mainButton.styling}
      />

      <Modal show={show} onHide={handleClose}>
        <Modal.Header className="hide-border padding-left" closeButton>
          <Modal.Title className="mt-3 larger-font">{title}</Modal.Title>
        </Modal.Header>
        <Modal.Body className="padding-left">{body()}</Modal.Body>
        <div className="mx-auto mb-5">
          <Button
            styling={confirmButton.styling}
            onClick={handleConfirm}
            label={confirmButton.label}
          />
          <br />
          <Button
            styling={cancelButton.styling}
            onClick={handleClose}
            label={cancelButton.label}
          />
        </div>
      </Modal>
    </>
  );
}

ConfirmationPopup.propTypes = {
  modal: PropTypes.shape({
    show: PropTypes.bool.isRequired,
    handleShow: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleConfirm: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.func.isRequired
  }).isRequired,
  mainButton: propTypes.button,
  confirmButton: propTypes.button,
  cancelButton: propTypes.button
};
