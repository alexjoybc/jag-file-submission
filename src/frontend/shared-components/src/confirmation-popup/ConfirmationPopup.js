import React from "react";
import PropTypes from "prop-types";
import Modal from "react-bootstrap/Modal";
import { Button } from "../button/Button";

import "./ConfirmationPopup.css";

export default function ConfirmationPopup({
  modal: { show, handleShow, handleClose, handleConfirm, title, body1, body2 },
  mainButton: { mainLabel, mainStyling },
  confirmButton: { confirmLabel, confirmStyling },
  cancelButton: { cancelLabel, cancelStyling }
}) {
  return (
    <>
      <Button onClick={handleShow} label={mainLabel} styling={mainStyling} />

      <Modal show={show} onHide={handleClose}>
        <Modal.Header className="hide-border padding-left" closeButton>
          <Modal.Title className="mt-3 larger-font">{title}</Modal.Title>
        </Modal.Header>
        <Modal.Body className="padding-left">
          <p>{body1}</p>
          <p>{body2}</p>
        </Modal.Body>
        <div className="mx-auto mb-5">
          <Button
            styling={confirmStyling}
            onClick={handleConfirm}
            label={confirmLabel}
          />
          <br />
          <Button
            styling={cancelStyling}
            onClick={handleClose}
            label={cancelLabel}
          />
        </div>
      </Modal>
    </>
  );
}

ConfirmationPopup.propTypes = {
  modal: PropTypes.shape({
    show: PropTypes.bool.isRequired,
    handleShow: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleConfirm: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    body1: PropTypes.string.isRequired,
    body2: PropTypes.string
  }).isRequired,
  mainButton: PropTypes.shape({
    mainLabel: PropTypes.string.isRequired,
    mainStyling: PropTypes.string.isRequired
  }).isRequired,
  confirmButton: PropTypes.shape({
    confirmLabel: PropTypes.string.isRequired,
    confirmStyling: PropTypes.string.isRequired
  }).isRequired,
  cancelButton: PropTypes.shape({
    cancelLabel: PropTypes.string.isRequired,
    cancelStyling: PropTypes.string.isRequired
  }).isRequired
};
