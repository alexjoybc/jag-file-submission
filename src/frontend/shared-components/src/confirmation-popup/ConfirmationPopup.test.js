import React from "react";
import { mount } from "enzyme";
import EnzymeToJson from "enzyme-to-json";

import { ConfirmationPopup } from "./ConfirmationPopup";

describe("ConfirmationPopup Component", () => {
  const modal = {
    show: true,
    handleShow: jest.fn(),
    handleClose: jest.fn(),
    title: "Cancel process?",
    body1: "Your process will end.",
    body2: "You will be redirected back to the original application."
  };
  const mainButton = {
    mainLabel: "main label",
    mainStyling: "normal-blue btn"
  };
  const confirmButton = {
    confirmLabel: "Yes, cancel my process please",
    confirmStyling: "normal-blue btn consistent-width"
  };
  const cancelButton = {
    cancelLabel: "No, resume my process please",
    cancelStyling: "normal-white btn consistent-width"
  };

  const component = (
    <ConfirmationPopup
      modal={modal}
      mainButton={mainButton}
      confirmButton={confirmButton}
      cancelButton={cancelButton}
    />
  );

  test("Matches the snapshot", () => {
    const subject = mount(component);
    expect(EnzymeToJson(subject)).toMatchSnapshot();
  });
});
