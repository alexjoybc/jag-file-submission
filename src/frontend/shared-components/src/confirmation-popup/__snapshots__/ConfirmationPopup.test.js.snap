// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPopup Component First child matches the snapshot 1`] = `
<div>
  <button
    class="normal-blue btn"
    type="button"
  >
    main label
  </button>
</div>
`;

exports[`ConfirmationPopup Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="normal-blue btn"
      type="button"
    >
      main label
    </button>
  </div>
</DocumentFragment>
`;

exports[`ConfirmationPopup Component Matches the snapshot 2`] = `
<div>
  <button
    class="normal-blue btn"
    type="button"
  >
    main label
  </button>
</div>
`;

exports[`ConfirmationPopup Component Second child matches the snapshot 1`] = `undefined`;
